olive/data/component/dataset.py

```diff
@@ -60,6 +61,53 @@ def to_snpe_dataset(self):
         """
         pass
 
+    def to_hf_dataset(self, label_name=None):
+        """
+        This function is used to convert the dataset to huggingface dataset
+
+        :param label_name: The name of the label column in the new dataset
+        """
+        from datasets import Dataset
+
+        # name for label column
+        label_cols = ["labels"]
+        if hasattr(self, "label_cols"):
+            label_cols = self.label_cols or ["labels"]  # some children classes may not have label_cols
+        label_name = label_name or label_cols[0]
+
+        if hasattr(self, "data") and isinstance(self.data, Dataset):
+            # deepcopy the dataset since we might modify it
+            hf_dataset = deepcopy(self.data)
+            for col_name in label_cols[1:]:
+                # remove the other label columns
+                # label_cols is a list but we only use the first element for now
+                hf_dataset.remove_columns(col_name)
+            # rename the label column
+            if label_cols[0] != label_name:
+                if label_name in hf_dataset.column_names:
+                    raise ValueError(f"Cannot rename label column to {label_name} since it already exists")
+                hf_dataset = hf_dataset.rename_column(label_cols[0], label_name)
+            if self.max_samples is not None:
+                # truncate the dataset if max_samples is not None
+                hf_dataset = hf_dataset.select(range(len(self)))
```

Comment:
Looks like it will be fast since it will just do slicing for range 
![image](https://github.com/microsoft/Olive/assets/94929125/63fea4f9-bed9-45c9-ba1b-4b64dfe2ee5a)
Will update this.

Reply:
Done!